src/Microsoft.AspNet.Mvc.Core/TempDataDictionary.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.AspNet.Hosting;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <inheritdoc />
+    public class TempDataDictionary : ITempDataDictionary
+    {
+        private Dictionary<string, object> _data;
+        private bool _loaded;
+        private ITempDataProvider _provider;
```

Comment:
I'm guessing some of these could/should be `readonly`? Like the provider and context accessor?